src/cli.erl

```diff
@@ -0,0 +1,104 @@
+-module(cli).
+
+-include("equivchecker.hrl").
+
+-export([run/1]).
+
+% TODO Refactor this monstrosity into something manageable
+-spec handler(map()) -> none().
+handler(#{target := Target, source := Source, json := Json, commit := Commit, stats := Stats}) when Commit ->
+    if not Json -> io:format("Checking commit ~p against commit ~p~n", [Target, Source]); true -> ok end,
+    {ok, ProjFolder} = file:get_cwd(),
+    OrigRepo = repo:copy(ProjFolder, ?ORIGINAL_SOURCE_FOLDER),
+    repo:checkout(OrigRepo, Target),
+    RefacRepo = repo:copy(ProjFolder, ?REFACTORED_SOURCE_FOLDER),
+    repo:checkout(RefacRepo, Source),
+    Res = check_equiv:check_equiv(filename:absname(OrigRepo), filename:absname(RefacRepo)),
+    show_result(Res, Json, Stats);
+handler(#{target := Target, source := Source, json := Json, commit := Commit, stats := Stats}) when not Commit ->
+    if not Json -> io:format("Checking folder ~p against folder ~p~n", [Target, Source]); true -> ok end,
+    Res = check_equiv:check_equiv(filename:absname(Target), filename:absname(Source)),
+    show_result(Res, Json, Stats);
+handler(#{target := Target, json := Json, commit := Commit, stats := Stats}) when Commit ->
+    if not Json -> io:format("Checking current folder against commit ~p~n", [Target]); true -> ok end,
+    {ok, ProjFolder} = file:get_cwd(),
+    Repo = repo:copy(ProjFolder, ?ORIGINAL_SOURCE_FOLDER),
+    repo:checkout(Repo, Target),
+    Res = check_equiv:check_equiv(filename:absname(ProjFolder), filename:absname(Repo)),
+    show_result(Res, Json, Stats);
+handler(#{target := Target, json := Json, commit := Commit, stats := Stats}) when not Commit ->
+    if not Json -> io:format("Checking current folder against ~p~n", [Target]); true -> ok end,
+    {ok, ProjFolder} = file:get_cwd(),
+    Res = check_equiv:check_equiv(filename:absname(ProjFolder), filename:absname(Target)),
+    show_result(Res, Json, Stats);
+handler(#{json := Json, commit := _, stats := Stats}) ->
+    if not Json -> io:format("Checking current folder against current commit~n"); true -> ok end,
+    {ok, ProjFolder} = file:get_cwd(),
+    Commit = repo:current_commit(),
+    Repo = repo:copy(ProjFolder, ?ORIGINAL_SOURCE_FOLDER),
+    repo:checkout(Repo, Commit),
+    Res = check_equiv:check_equiv(filename:absname(ProjFolder), filename:absname(Repo)),
+    show_result(Res, Json, Stats).
+
+setup() ->
+    % Sets the name of the master node
+    net_kernel:start(master, #{name_domain => shortnames}),
+
+    % These are needed for storing statistics
+    ets:new(stat, [named_table, public, set, {keypos, 1}]),
+    ets:insert(stat, {counts, []}),
+
+    % Creating temporary folders for the source and bytecode
+    file:make_dir(?TEMP_FOLDER),
+    file:make_dir(?ORIGINAL_BIN_FOLDER),
+    file:make_dir(?REFACTORED_BIN_FOLDER),
+
+    application:start(wrangler).
+
+cleanup() ->
+    ets:delete(stat),
+    file:del_dir_r(?TEMP_FOLDER), % TODO Handle error
```

Comment:
Or just put it under a `catch`.

Reply:
My plan was to deal with these kind of errors in the battle testing phase.
And I'm not even sure this should be checked explicitly.
Handling the case when the creation if this temp folder fails would make more sense.

Reply:
Alright. I'd suggest that you don't leave these TODO comments hanging around in PRs but turn them into proper issues/feature requests before the merge.